{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"init"},{"kind":"text","text":"("},{"kind":"externalParam","text":"errnoCode"},{"kind":"text","text":" "},{"kind":"internalParam","text":"code"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"CInt","preciseIdentifier":"s:s4CInta"},{"kind":"text","text":", "},{"kind":"externalParam","text":"function"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"StaticString","preciseIdentifier":"s:s12StaticStringV"},{"kind":"text","text":")"}],"languages":["swift"],"platforms":["macOS"]}]},{"kind":"parameters","parameters":[{"name":"errorCode","content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"the "},{"type":"codeVoice","code":"errno"},{"type":"text","text":" that was set for the operation."}]}]},{"name":"function","content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"The function the error happened in, the human readable description will be generated automatically when needed."}]}]}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/nio\/ioerror\/init(errnocode:function:)"],"traits":[{"interfaceLanguage":"swift"}]}],"identifier":{"url":"doc:\/\/org.swift.docc\/documentation\/NIO\/IOError\/init(errnoCode:function:)","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"Creates a new `IOError``"}],"kind":"symbol","metadata":{"role":"symbol","title":"init(errnoCode:function:)","roleHeading":"Initializer","fragments":[{"kind":"identifier","text":"init"},{"kind":"text","text":"("},{"kind":"externalParam","text":"errnoCode"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"CInt","preciseIdentifier":"s:s4CInta"},{"kind":"text","text":", "},{"kind":"externalParam","text":"function"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"StaticString","preciseIdentifier":"s:s12StaticStringV"},{"kind":"text","text":")"}],"symbolKind":"init","externalID":"s:3NIO7IOErrorV9errnoCode8functionACs5Int32V_s12StaticStringVtcfc","modules":[{"name":"NIO"}],"platforms":[]},"hierarchy":{"paths":[["doc:\/\/org.swift.docc\/documentation\/NIO","doc:\/\/org.swift.docc\/documentation\/NIO\/IOError"]]},"references":{"doc://org.swift.docc/documentation/NIO/IOError/init(errnoCode:function:)":{"role":"symbol","title":"init(errnoCode:function:)","fragments":[{"kind":"identifier","text":"init"},{"kind":"text","text":"("},{"kind":"externalParam","text":"errnoCode"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"CInt","preciseIdentifier":"s:s4CInta"},{"kind":"text","text":", "},{"kind":"externalParam","text":"function"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"StaticString","preciseIdentifier":"s:s12StaticStringV"},{"kind":"text","text":")"}],"abstract":[{"type":"text","text":"Creates a new `IOError``"}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/IOError\/init(errnoCode:function:)","kind":"symbol","type":"topic","deprecated":true,"url":"\/documentation\/nio\/ioerror\/init(errnocode:function:)"},"doc://org.swift.docc/documentation/NIO":{"role":"collection","title":"NIO","abstract":[],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO","kind":"symbol","type":"topic","url":"\/documentation\/nio"},"doc://org.swift.docc/documentation/NIO/IOError":{"role":"symbol","title":"IOError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"IOError"}],"abstract":[{"type":"text","text":"An "},{"type":"codeVoice","code":"Error"},{"type":"text","text":" for an IO operation."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/IOError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"IOError"}],"url":"\/documentation\/nio\/ioerror"}}}